src/openvic-simulation/history/CountryHistory.hpp

```diff
@@ -36,12 +40,25 @@ namespace OpenVic {
 		std::optional<GovernmentType const*> PROPERTY(government_type);
 		std::optional<fixed_point_t> PROPERTY(plurality);
 		std::optional<NationalValue const*> PROPERTY(national_value);
-		std::optional<bool> PROPERTY(civilised);
+		std::optional<bool> PROPERTY_CUSTOM_PREFIX(civilised, is);
 		std::optional<fixed_point_t> PROPERTY(prestige);
 		std::vector<Reform const*> PROPERTY(reforms);
 		std::optional<Deployment const*> PROPERTY(inital_oob);
-		// TODO: technologies, tech schools, and inventions when PR#51 merged
-		// TODO: starting foreign investment
+		std::optional<TechnologySchool const*> PROPERTY(tech_school);
+		std::map<Technology const*, bool> PROPERTY(technologies);
+		std::map<Invention const*, bool> PROPERTY(inventions);
+		fixed_point_map_t<Country const*> PROPERTY(foreign_investment);
+		std::optional<fixed_point_t> PROPERTY(consciousness);
+		std::optional<fixed_point_t> PROPERTY(nonstate_consciousness);
+		std::optional<fixed_point_t> PROPERTY(literacy);
+		std::optional<fixed_point_t> PROPERTY(nonstate_culture_literacy);
+		std::optional<bool> PROPERTY_CUSTOM_PREFIX(releasable_vassal, is);
+		std::optional<fixed_point_t> PROPERTY(colonial_points);
+		std::unordered_set<std::string> PROPERTY(country_flags);
+		std::unordered_set<std::string> PROPERTY(global_flags);
```

Comment:
```suggestion
		string_set_t PROPERTY(country_flags);
		string_set_t PROPERTY(global_flags);
```
Although the order here probably doesn't matter, I'd prefer to stay on the safe side and use `string_set_t` that guaruntees the same order across all machines. Could you also change the `std::unordered_set`s in `Technology.hpp` and `Invention.hpp` to `std::set`s?.

Reply:
sure thing!